src/trackinfoobject.cpp

```diff
@@ -738,23 +739,51 @@ QString TrackInfoObject::getURL() {
 }
 
 ConstWaveformPointer TrackInfoObject::getWaveform() {
+    QMutexLocker lock(&m_qMutex);
     return m_waveform;
 }
 
 void TrackInfoObject::setWaveform(ConstWaveformPointer pWaveform) {
+    if (m_waveform == pWaveform) {
+        return;
+    }
+
+    QMutexLocker lock(&m_qMutex);
     m_waveform = pWaveform;
+    markDirtyAndUnlock(&lock);
     emit(waveformUpdated());
 }
 
 ConstWaveformPointer TrackInfoObject::getWaveformSummary() const {
+    QMutexLocker lock(&m_qMutex);
     return m_waveformSummary;
 }
 
 void TrackInfoObject::setWaveformSummary(ConstWaveformPointer pWaveform) {
+    if (m_waveformSummary == pWaveform) {
```

Comment:
Here you rely on the thread-safety guarantees of QSharedPointer. Perfectly ok, no need for locking :)

But we should apply those rules consistently and remove the extra locks from the getters (see above)